packages/ra-core/src/auth/useLogoutIfAccessDenied.ts

```diff
@@ -103,12 +103,17 @@ const useLogoutIfAccessDenied = (): LogoutIfAccessDenied => {
                     if (logoutUser) {
                         logout({}, redirectTo);
                     } else {
-                        if (redirectTo.startsWith('http')) {
-                            // absolute link (e.g. https://my.oidc.server/login)
-                            window.location.href = redirectTo;
+                        if (redirectTo != null) {
+                            if (redirectTo.startsWith('http')) {
+                                // absolute link (e.g. https://my.oidc.server/login)
+                                window.location.href = redirectTo;
+                            } else {
+                                // internal location
+                                navigate(redirectTo);
+                            }
                         } else {
-                            // internal location
-                            navigate(redirectTo);
+                            // If there is not redirect return false saying loguout is not successful
+                            return false;
```

Comment:
I think it would be better instead of using this nested if else condition we can make functions and use. This will help others to understand more easily and help to maintain the clean code e.g:

```

`const handleRedirect = (url) => {
        if (url.startsWith('http')) {
            window.location.href = url; // Redirect to external URL
        } else {
            navigate(url); // Navigate to internal route
        }
    };



    if (logoutUser) {
        // Perform the logout operation
        logout({}, redirectTo);
    } else if (redirectTo) {
        handleRedirect(redirectTo);
    } else {
        // No redirect specified; log and return false
        console.error('Logout unsuccessful: No redirect URL provided.');
        return false;
    }
};`
`````